test/modules/random_test.py

```diff
@@ -0,0 +1,73 @@
+import sqlite3
+import unittest
+from unittest.mock import MagicMock
+import random
+
+from irc.client import Event, NickMask
+
+import bavi.bot
+import bavi.modules.random as random_module
+
+class RandomModuleTestCase(unittest.TestCase):
+    def setUp(self):
+        self.bot = bavi.bot.Bot(None)
+        random_module.init(self.bot)
+        self.bot.say = MagicMock()
+        self.bot.reply_to = MagicMock()
+        random.seed(0)
+        
+    def test_random_make_list(self):
+        commas = random_module.make_list("1,2,3,4,5,6,7,8,9,0")
+        bar = random_module.make_list("a|b|c|d|e")
+        forward_slash = random_module.make_list("b/d/e/f/g")
+        back_slash = random_module.make_list("q\\v\\e\\3\\f\\b\\g\\g")
+        mixed = random_module.make_list("q,v,e\\3\\f\\b,g\\g\\a|b|c|d|e")
+        nothing = random_module.make_list("")
+        none_check = random_module.make_list(None)
+
+        self.assertEqual(len(commas), 10)
+        self.assertEqual(len(bar), 5)
+        self.assertEqual(len(forward_slash), 5)
+        self.assertEqual(len(back_slash), 8)
+        self.assertEqual(len(mixed), 6)
+        self.assertEqual(len(nothing), 1)
+        self.assertEqual(len(none_check), 0)
+        
+        self.assertEqual(commas[0], '1')
+        self.assertEqual(mixed[4], 'g')
+        self.assertEqual(nothing[0], '')
+        self.assertEqual(mixed[4], 'g')
+        
+    def test_random_normal(self):
+        self.bot.on_pubmsg(
+                None,
+                Event(
+                    'pubmsg',
+                    NickMask('user!ident@host'),
+                    '#test',
+                    ['.choose 1,2,3,4,5']
+                )
+        )
+        
+        self.bot.reply_to.assert_called_with(
+                NickMask('user!ident@host'),
+                '#test',
+                'Your choices: [\'1\', \'2\', \'3\', \'4\', \'5\'], I chose: 4.'
```

Comment:
I know you did `seed(0)`, but I'm wondering does Python actually guarantee that `random.choice` is deterministic by the seed?

Reply:
Looking at `random.py`, `random.choice` uses `_randbelow`, which uses the `self.random` object, which is modified by the `seed` call.

That and in my testing I was unable to get the results to change with a seed of zero.

https://github.com/python/cpython/blob/master/Lib/random.py#L254

Reply:
Sure, I had assumed you had tried it, but I was wondering if the documentation guarantees this or if it's an implementation detail of CPython.  Not that I plan to run this on PyPy, but it's nice to be portable.

What I ended up doing for the time module was grabbing the call args manually and checking the beginning of the string (ignoring the timestamp from `datetime.now()` which is also somewhat "random" from the point of view of a test)

Reply:
Ah I see what you are saying. I'll see if I can override the call with Mock to make it more portable.

I already implemented your other changes in my branch as well. I'll polish that up later this evening.